psi4/driver/driver_findif.py

```diff
@@ -1017,65 +1126,371 @@ def init_string(data):
     return _process_hessian(H_pi, B_pi, massweighter, data["print_lvl"])
 
 
-########
+class FiniteDifferenceComputer(BaseComputer):
 
-# This function is shifting `_energy_is_invariant` from driver to findif but isn't in final DDD form.
-def prep_findif(mol, irrep, mode, gradient=None):
+    molecule: Any
+    driver: DriverEnum
+    metameta: Dict[str, Any] = {}
+    task_list: Dict[str, BaseComputer] = {}
+    findifrec: Dict[str, Any] = {}
+    computer: BaseComputer = AtomicComputer
+    method: str
 
-        findif_stencil_size = core.get_option("FINDIF", "POINTS")
-        findif_step_size = core.get_option("FINDIF", "DISP_SIZE")
+    @validator('driver')
+    def set_driver(cls, driver):
+        egh = ['energy', 'gradient', 'hessian']
+        if driver not in egh:
+            raise ValidationError(f"""Wrapper is unhappy to be calling function ({driver}) not among {egh}.""")
 
-        translations_projection_sound = (not core.get_option('SCF', 'EXTERN')
+        return driver
+
+    @validator('molecule')
+    def set_molecule(cls, mol):
+        mol.update_geometry()
+        mol.fix_com(True)
+        mol.fix_orientation(True)
+        return mol
+
+    def __init__(self, **data):
```

Comment:
A comment on `data` would be good. It looks like this is a massive list of keyword arguments. Some keyword arguments control findif operation, while others control the individual electronic structure singlepoints.

Reply:
I've added some notes, but these classes are still a bit of a free-for-all. All three pydantic classes need two passes: one to formalize the needed fields and add docstrings. this has happened for the manybody class. second is to forbid extra fields (see Config of base BaseComputer to see how) and find out what's getting attached and either formalize them or otherwise handle better. After those passes (or with them) a formal qcschema model for the procedures should be formed, then the Computer becomes fairly light around the Input/Result (roughly akin to AtomicComputer vs AtomicInput/AtomicResult). The pydantic objects here are fairly primitive, as they were written before the qcel ones got started, much less got developed, so these have some catching-up to do.